StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1516CodeFixProvider.cs

```diff
@@ -94,5 +95,37 @@ private SyntaxNode GetRelevantNode(SyntaxNode innerNode)
 
             return null;
         }
+
+        private class FixAll : DocumentBasedFixAllProvider
+        {
+            public static FixAllProvider Instance { get; } =
+                new FixAll();
+
+            protected override string CodeActionTitle =>
+                LayoutResources.SA1516CodeFix;
+
+            protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document)
+            {
+                var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+                var syntaxRoot = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+                List<SyntaxNode> nodes = new List<SyntaxNode>();
+
+                foreach (var diagnostic in diagnostics)
+                {
+                    var node = syntaxRoot.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                    node = GetRelevantNode(node);
```

Comment:
:exclamation: This treatment of `GetRelevantNode` is different from the one above.

* If `GetRelevantNode` can return `null`, any such nodes should not be added to `nodes`
* If `GetRelevantNode` cannot return `null`, the code above (in the single-fix case) should use `.` instead of `?.` to access its members

:bulb: In addition, it seems clear that we do not need to execute `FindNode` inside of `RegisterCodeFixesAsync`. You can move this call into `GetTransformedDocumentAsync` for the single-fix case to maximize the performance of `RegisterCodeFixesAsync`. Note that if you don't do this, you're going to execute `FindNode` twice for every diagnostic during a fix all operation - the first time is used to determine the equivalence keys (in the call to `RegisterCodeFixesAsync`), and the second time is used by the `FixAllInDocumentAsync` operation.